server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java

```diff
@@ -76,8 +84,155 @@ public LogicalPlanVisitor(Analysis analysis) {
 
     @Override
     public PlanNode visitQuery(QueryStatement queryStatement, MPPQueryContext context) {
-      // TODO: Generate logical planNode tree for query statement
-      return null;
+      PlanBuilder planBuilder = planSelectComponent(queryStatement);
+
+      if (queryStatement.getWhereCondition() != null) {
+        planBuilder =
+            planQueryFilter(planBuilder, queryStatement.getWhereCondition().getQueryFilter());
+      }
+
+      if (queryStatement.isGroupByLevel()) {
+        planBuilder =
+            planGroupByLevel(
+                planBuilder,
+                ((AggregationQueryStatement) queryStatement).getGroupByLevelComponent());
+      }
+
+      if (queryStatement instanceof FillQueryStatement) {
+        planBuilder =
+            planFill(planBuilder, ((FillQueryStatement) queryStatement).getFillComponent());
+      }
+
+      planBuilder = planFilterNull(planBuilder, queryStatement.getFilterNullComponent());
+      planBuilder = planSort(planBuilder, queryStatement.getResultOrder());
+      planBuilder = planLimit(planBuilder, queryStatement.getRowLimit());
+      planBuilder = planOffset(planBuilder, queryStatement.getRowOffset());
+      return planBuilder.getRoot();
+    }
+
+    private PlanBuilder planSelectComponent(QueryStatement queryStatement) {
```

Comment:
Besides the series in ResultColumn, the related series in QueryFilter should also be considered here.

Reply:
Sure, let's add a TODO here

Reply:
added